AudioEncode/CMakeLists.txt

```diff
@@ -1,6 +1,13 @@
-set(HEADERS)
+set(HEADERS
+  audio_encode.h)
```

Comment:
Don't add headers into CMake project, cmake should handle them as target dependency automatically.

Reply:
> Don't add headers into CMake project, cmake should handle them as target dependency automatically.

This is an incorrect use of `target_include_directories`. Each directory added to `target_include_directories` is a search path for `#include` directives which means nothing is actually added and therefore it will not show up in any IDE.

Reply:
@winterheart Fixing the situation with IDEs (especially my own) is the entire purpose of doing this. I don't know why you are a so dead set against listing the header files but this is something I require. I have done everything else you have asked, please remove your block on this merge.

Reply:
Try here is a mix of cmakelists with header files set this way already. I fail to see the harm in consistency.

Reply:
Depends on the IDE and how you use it. e.g. I believe this is true if you use CMake to spit out a .vcproj, but if you open the folder directly in VS (Ninja etc) it finds the headers on its own.

Reply:
> @JeodC: There is a mix of cmakelists with header files set this way already.

Good point.

> @SiriusTR: Depends on the IDE and how you use it.

An IDE **should not** display include directories specifically because it could be a system directory which could easily have hundreds of header files. They may go to great lengths to try and only include what it believes is not a system directory but it's impossible to know and based on assumptions about your system.  The proper way is to display files explicitly indicated as being part of the project. I see no reason to oppose proper header file inclusion.

Reply:
Trying to follow this conversation here, so bare with me. First off, we shouldn't care about what IDE is used, when this project can be edited in something simple as VI and built just from command line, so we only care about the cmake structure here.  I am used to cmake projects with cmake submodules, and in those submodules, their have their own set of header files, and perhaps even their own `include` directory too.  I think when one submodule wants to include another, just needs to add that cmake defined submodule into their dep list.

Reply:
> we shouldn't care about what IDE is used, when this project can be edited in something simple as VI and built just from command line,

I'm not modifying it for a specific IDE just _an_ IDE, any IDE. While not technically required for the build process, it also does not harm the build process. We should be encouraging people to work on the project rather than discouraging it. IDEs are tools to help developers so going out of your way to make it more difficult for developers to work on your project is antithetical to being open source in the first place.

If you look at the PR, the only headers I've listed are the ones in the same directory that are used by their implementation file counterparts. I have not listed headers from other submodules, only the current one.

Reply:
@GravisZro we requested changes to your contribution. If you disagree with them without a good enough reason, then PR is probably not going to be merged. Enumerating headers in cmake project is undesirable practice that prevents to detect unused headers on some IDEs (for example).

Earlier I suggested update your particular QtCreator to recent version, where your issue does not exists.

Reply:
@winterheart 
> IDEs are tools to help developers so going out of your way to make it more difficult for developers to work on your project is antithetical to being open source in the first place.

How is this not a good enough reason to keep the header files listed? Why are you being so uncompromising?

Reply:
> > we shouldn't care about what IDE is used, when this project can be edited in something simple as VI and built just from command line,
> 
> I'm not modifying it for a specific IDE just _an_ IDE, any IDE. While not technically required for the build process, it also does not harm the build process. We should be encouraging people to work on the project rather than discouraging it. IDEs are tools to help developers so going out of your way to make it more difficult for developers to work on your project is antithetical to being open source in the first place.
> 
> If you look at the PR, the only headers I've listed are the ones in the same directory that are used by their implementation file counterparts. I have not listed headers from other submodules, only the current one.

As I said, we will just follow the best cmake standards, where it is expected some IDE with cmake support will also follow the same suit.  Most of the cmake based C++ projects on GitHub just follow that basic standard, so the source can be built with just CLI as a minimal.  People encouraged working on Descent 3 will be the ones that love both the game and C++.

Reply:
> where it is expected some IDE with cmake support will also follow the same suit.

Except cmake is a build system, not a project management system. If there were a project file then I would use that but there is only cmake files.

> Most of the cmake based C++ projects on GitHub just follow that basic standard, so the source can be built with just CLI as a minimal. 

Literally nothing I'm proposing will interfere with cmake building D3.

The question is, why is this small compromise unacceptable?

Reply:
I didn't say the changes were unacceptable, still need re-review them again.  I wanted to kick off a cmake discussion, to get a feel with everyone on best direction to go.  When everyone is on the same page, I hope that PRs aren't delayed by any further arguments.

Cmake is a build system, but it has some project generators, then there are IDEs with the ability to establish the project structure based on cmake info.

Reply:
I did not step in before because I thought things would sort themselves out.

Most projects I've seen list headers in the source files, especially for IDE support, it's not a big pain to do it. You add them in the target sources like you do for cpps. CMake can do without, but it's not wrong at all to do it. Everything I found online about this issue corroborates this [1](https://stackoverflow.com/questions/13703647/how-to-properly-add-include-directories-with-cmake) [2](https://stackoverflow.com/questions/61057405/cmake-add-header-files-to-target). Craig Scott (author of the main CMake book) suggests [file sets](https://discourse.cmake.org/t/target-include-directories-vs-headers-in-target-sources/2625/2) as an alternative, that's something we can look into later.

tl;dr I think this change is totally acceptable.